### PR TITLE
type: fix type errors of React.Key

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close: #45017 

1. 文档错误提交：https://github.com/ant-design/ant-design/pull/37144/files#diff-0bdf37a1fc0ec124fb3aa391932eaa134bd1de8c6f766f5e4082edbc30ce6c42

2. 修复 `React.Key` TS 类型错误 cherry-pick ：https://github.com/ant-design/ant-design/pull/44938

3. 更新 test 快照： cherry-pick： https://github.com/ant-design/ant-design/pull/44989

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |      --     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 913f1c3</samp>

Fix formatting of `components/tree/index.zh-CN.md` to improve readability of the documentation.

### <samp>🤖 Generated by Copilot at 913f1c3</samp>

* Add `checkable` prop to `Tree` and `DirectoryTree` components to support rendering checkboxes for tree nodes (7,7)

### CODE REVIEW

**Review by MadCcc — APPROVED** [review]

**Review by kiner-tang — APPROVED** [review]

**Comment by Wxh16144** [comment]
需要重新 review~

**Comment by Wxh16144** [comment]
@afc163 是否需要 pick https://github.com/react-component/field-form/pull/617 开一个 `1.34.x` 分支给 `4.x` 用？

File: package.json [hunk 1]
```diff
@@ -131,7 +131,7 @@
     "rc-dialog": "~9.0.2",
     "rc-drawer": "~6.3.0",
     "rc-dropdown": "~4.0.1",
-    "rc-field-form": "~1.34.2",
+    "rc-field-form": "~1.38.2",
```
Comment (afc163) [inline package.json:134]: 这个更新是否会影响 Form 的逻辑？

Comment (afc163) [inline package.json:134]: 是否引入了 changelog 里未提到的新特性？

Comment (yoyo837) [reply package.json:134]: 见群消息, 4.x 的 `rc-field-form` 之前从 1.30直接拉到1.34的, 当时和5.x拉平了, 想着, 如果条件允许, 干脆也直接再次拉平得了. @zombieJ 来看看

Comment (zombieJ) [reply package.json:134]: 拉平问题不大，从上次拉平起，我们其实就没有在文档里提过新特性。Form 组件本身的 TS 定义也没有做过透传。

**Pull request merged**